### PR TITLE
Updates regular user login functionality to include sad paths

### LINKED ISSUE #25: User Story 14, User cannot log in with bad credentials
As a visitor
When I visit the login page ("/login")
And I submit invalid information
Then I am redirected to the login page
And I see a flash message that tells me that my credentials were incorrect
I am NOT told whether it was my email or password that was incorrect

### LINKED ISSUE #26: User Story 13, User can Login
As a visitor
When I visit the login path
I see a field to enter my email address and password
When I submit valid information
If I am a regular user, I am redirected to my profile page
If I am a merchant user, I am redirected to my merchant dashboard page
If I am an admin user, I am redirected to my admin dashboard page
And I see a flash message that I am logged in

### PR DESCRIPTION
**Pull request by rer7891**
## What functionality does this accomplish?
closes #26 
closes #25 

**Description:**
Updates regular user functionality to include sad paths for wrong email(email cannot be found in the database) or the password does not match the account.
## What did you struggle on to complete?
 How to validate the password against the email account. 

### Test Coverage Percentage: x%
- [ ] No Tests have been changed
- [x] Some Tests have been changed: Login under feature users was updated to include sad paths
- [ ] All of the Tests have been changed(Please describe what in the world happened):
- [ ] New Tests added for new features

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code
- [ ] I have partially tested my code (please explain why):

## Helpful Resources:
* Resource link AND small description of info gathered/learned

## Review Requests(optional):
My flash message for wrong inputs is: "Sorry Invalid Password or Email." and the user story says: "I see a flash message that tells me that my credentials were incorrect I am NOT told whether it was my email or password that was incorrect" I hope this meets this specification. I believe it does.
images

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -5,8 +5,13 @@ def new
 
   def create
     user = User.find_by(email: params[:email])
-    session[:user_id] = user.id
-    flash[:success] = "Welcome #{user.email}"
-    redirect_to "/profile"
+    if user != nil && user.authenticate(params[:password])
```
Comment (wrenisit) [inline app/controllers/sessions_controller.rb]: Great job on the authenticate!

File: app/models/order.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Order <ApplicationRecord
-  validates_presence_of :name, :address, :city, :state, :zip
+  validates_presence_of :name, :address, :city, :state, :zip, numericality: { only_integer: true, message: "must be a number" }
```
Comment (wrenisit) [inline app/models/order.rb]: Wow! Great reach here with numercality!

File: spec/features/users/login_spec.rb [hunk 3]
```diff
@@ -28,6 +27,42 @@
...
+
+      fill_in :email, with: "yo@gmail.com"
+      fill_in :password, with: "notsecure123"
+      fill_in :password_confirmation, with: "notsecure123"
+      click_button "Log In"
+
+      expect(current_path).to eq("/login")
+      expect(page).to have_content("Sorry Invalid Password or Email.")
+    end
+
```
Comment (wrenisit) [inline spec/features/users/login_spec.rb]: The sad path testing for both bad email and bad password is awesome! Great coverage.

File: spec/features/users/login_spec.rb [hunk 4]
```diff
@@ -28,6 +27,42 @@
       expect(page).to have_content("Welcome go@foogle.com")
     end
 
+    it "won't login if password does not match" do
+
+      new_user = create(:regular_user, email: "yo@gmail.com", password: "123password", password_confirmation: "123password")
```
Comment (jfangonilo) [inline spec/features/users/login_spec.rb]: Glad to see FactoryBot use

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@ class ApplicationController < ActionController::Base
   helper_method :cart, :current_user
 
   def current_user
-    User.find(session[:user_id])
+  @user ||= User.find(session[:user_id]) if session[:user_id]
```
Comment (jfangonilo) [inline app/controllers/application_controller.rb]: Potential for refactor here, just not sure exactly how :)

File: app/models/user.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@ class User <ApplicationRecord
   validates_presence_of :address, presence: true
   validates_presence_of :city, presence: true
   validates_presence_of :state, presence: true
-  validates_presence_of :zip, presence: true
+  validates_presence_of :zip, presence: true, numericality: { only_integer: true, message: "must be a number" }
```
Comment (jfangonilo) [inline app/models/user.rb]: Could change syntax here to match the rest of our validations. Maybe something like `validates_numericality_of :zip, only_integer: true`.  I'm also not entirely sure why presence: true is added on the end of the presence validations... seems kind of redundant but I'm not entirely clear on it. Maybe table it for now if it's working

**Comment by jfangonilo** [comment]
Oh, and also please make sure to include test coverage moving forward

**Pull request merged**